testing/src/test/java/io/stargate/it/http/graphql/schemafirst/DeleteTest.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.it.http.graphql.schemafirst;
+
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+
+import com.datastax.oss.driver.api.core.CqlIdentifier;
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.cql.ResultSet;
+import com.jayway.jsonpath.JsonPath;
+import io.stargate.it.driver.CqlSessionExtension;
+import io.stargate.it.driver.TestKeyspace;
+import io.stargate.it.http.RestUtils;
+import io.stargate.it.storage.StargateConnectionInfo;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(CqlSessionExtension.class)
+public class DeleteTest extends GraphqlFirstTestBase {
+  private static GraphqlFirstClient CLIENT;
+  private static String NAMESPACE;
+  private static CqlSession SESSION;
+
+  @BeforeAll
+  public static void setup(
+      StargateConnectionInfo cluster, @TestKeyspace CqlIdentifier keyspaceId, CqlSession session) {
+    CLIENT =
+        new GraphqlFirstClient(
+            cluster.seedAddress(), RestUtils.getAuthToken(cluster.seedAddress()));
+    NAMESPACE = keyspaceId.asInternal();
+    SESSION = session;
+    CLIENT.deploySchema(
+        NAMESPACE,
+        "type Foo @cql_input {\n"
+            + "  pk: Int! @cql_column(partitionKey: true)\n"
+            + "  cc1: Int! @cql_column(clusteringOrder: ASC)\n"
+            + "  cc2: Int! @cql_column(clusteringOrder: ASC)\n"
+            + "}\n"
+            + "type DeleteFooResult @cql_payload {\n"
+            + "  applied: Boolean\n"
+            + "}\n"
+            + "type Query {\n"
+            + "  foo(pk: Int!, cc1: Int!, cc2: Int!): Foo\n"
+            + "}\n"
+            + "type DeleteFooResponse @cql_payload {\n"
+            + "  applied: Boolean"
+            + "}\n"
+            + "type Mutation {\n"
+            + "  deleteFoo(foo: FooInput!): Boolean \n"
+            + "  deleteFoo2(pk: Int, cc1: Int, cc2: Int): Boolean\n"
+            + "    @cql_delete(targetEntity: \"Foo\")\n"
```

Comment:
Because this has the `@cql_delete` directive should maybe this should be called `foo2` instead of `deleteFoo2`?

Reply:
The operation names are completely free-form. There is just the rule that if it starts with certain prefixes then you can skip the directive, but that's just a convenience.

Here we have to use the annotation because we want to provide the `targetEntity` argument.